planner/transformation/rule_FilterAggregateTranspose.go

```diff
@@ -0,0 +1,104 @@
+package transformation
+
+import (
+	"github.com/pingcap/tidb/expression"
+	plannercore "github.com/pingcap/tidb/planner/core"
+	"github.com/pingcap/tidb/planner/memo"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/set"
+)
+
+// FilterAggregateTransposeRule pushes the filters through the aggregate:
+// before: filter(aggregate(any), f1, f2, ...)
+// after:  aggregate(filter(any, f1, f2, ...)),      totaly push through
+//    or:  filter(aggregate(filter(any, f1)), f2), partialy push through
+type FilterAggregateTransposeRule struct {
+	baseTransform
+}
+
+// NewFilterAggregateTransposeRule creates a new FilterAggregateTransposeRule.
+func NewFilterAggregateTransposeRule() *FilterAggregateTransposeRule {
+	pattern := memo.BuildPattern(memo.OperandSelection, memo.BuildPattern(memo.OperandAggregation))
+	return &FilterAggregateTransposeRule{baseTransform{pattern}}
+}
+
+// OnTransform push the filters through the aggregate:
+// before: filter(aggregate(any), f1, f2, ...)
```

Comment:
Need we repeat these comments?
`OnTransform` only keeps `does the actual transformation`.
And leave the detailed comments to where the struct is defined.